components/renku_data_services/project/db.py

```diff
@@ -61,11 +59,23 @@ async def get_projects(
             _ = await session.connection()
             stmt = select(schemas.ProjectORM)
             stmt = stmt.where(schemas.ProjectORM.id.in_(project_ids))
+            if namespace:
+                stmt = (
+                    stmt.where(schemas.NamespaceORM.slug == namespace.lower())
+                    .where(schemas.ProjectSlug.namespace_id == schemas.NamespaceORM.id)
+                    .where(schemas.ProjectORM.id == schemas.ProjectSlug.project_id)
+                )
```

Comment:
thought: should we outsource this to a `namespace_slug_filter` function? also in `get_project_by_namespace_slug`

Reply:
Is this (b872772fa449a61eeef50b5572288c98ba29fdf5) what you had in mind?

Reply:
yes, perfect :+1: